extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -90,11 +90,12 @@ export class AmpFlyingCarpet extends AMP.BaseElement {
   }
 
   /** @override */
-  onLayoutMeasure() {
+  onMeasureChanged() {
     const width = this.getLayoutWidth();
-    this.getVsync().mutate(() => {
+    this.mutateElement(() => {
       setStyle(this.container_, 'width', width, 'px');
     });
+    this.scheduleLayout(dev().assertElement(this.container_));
```

Comment:
This needs to be reverted. It can only schedule layout for the container if it itself has been laid out.

Reply:
Can `onMeasureChanged` get called before `layout`?

Reply:
It can be called after build.

Reply:
https://github.com/ampproject/amphtml/pull/16816